src/background.js

```diff
@@ -34,29 +36,371 @@
+    }
+  },
+
+  /**
+  * saves a user preference for an origin, overriding
+  * the default setting.
+  * @param {String} userAction enum of block, cookieblock, noaction
+  * @param {String} origin the third party origin to take action on
+  */
+  saveAction: function(userAction, origin) {
+    var allUserActions = {'block': pb.USER_BLOCK,
+                          'cookieblock': pb.USER_COOKIE_BLOCK,
+                          'allow': pb.USER_ALLOW};
+    pb.storage.setupUserAction(origin, allUserActions[userAction]);
+    pb.log("Finished saving action " + userAction + " for " + origin);
+
+    // TODO: right now we don't determine whether a reload is needed
+    return true;
+  },
+
+  /**
+  * reloads a tab
+  * @param {Integer} tabId the chrome tab id
+  */
+  reloadTab: function(tabId){
+    chrome.tabs.reload(tabId);
+  },
+
+  /**
+  * Populate tabs object with currently open tabs when extension is updated or installed. 
+  */
+  updateTabList: function(){
+    // Initialize the tabData/frames object if it is falsey
+    pb.tabData = pb.tabData || {};
+    chrome.tabs.query({currentWindow: true, status: 'complete'}, function(tabs){
+      for(var i = 0; i < tabs.length; i++){
+        var tab = tabs[i];
+        pb.tabData[tab.id] = {
+          frames: {
+            0: {
+              parent: -1,
+              url: tab.url
+            }
+          },
+          trackers: {}
+        };
+      }
+    });
+  },
+
+  /**
+   * Initialize the Cookieblock List:
+   * * Download list form eff
+   * * Merge with existing cookieblock list if any 
+   * * Add any new domains to the action map
+   * Set a timer to update every 24 hours
+   **/
+  initializeCookieBlockList: function(){
+    pb.updateCookieBlockList();
+    setInterval(pb.updateCookieBlockList, pb.utils.oneDay());
+  },
+
+  /**
+  * Update the cookie block list with a new list
+  * add any new entries that already have a parent domain in the action_map
+  * and remove any old entries that are no longer in the cookie block list
+  * from the action map
+  **/
+  updateCookieBlockList: function(){
+    pb.utils.xhrRequest(pb.COOKIE_BLOCK_LIST_URL, function(err,response){
+      if(err){
+        console.error('Problem fetching privacy badger policy hash list at', 
+                  pb.COOKIE_BLOCK_LIST_URL, err.status, err.message);
+        return;
+      }
+      var cookieblock_list = pb.storage.getBadgerStorageObject('cookieblock_list');
+      var action_map = pb.storage.getBadgerStorageObject('action_map');
+
+      var newCbDomains = _.map(response.split("\n"), function(d){ return d.trim();});
+      var oldCbDomains = Object.keys(cookieblock_list.getItemClones());
+
+      var addedDomains = _.difference(newCbDomains, oldCbDomains);
+      var removedDomains = _.difference(oldCbDomains, newCbDomains);
+      pb.log('adding to cookie blocklist:', addedDomains);
+      pb.log('removing from cookie blocklist:', removedDomains);
```

Comment:
should this be formatted in some way? in python it'd be something like ", ".join(removedDomains)